packages/core/src/__tests__/state-manager.test.ts

```diff
@@ -233,13 +233,13 @@ describe('rewind', () => {
   });
 
   test('return read-only snapshot', async () => {
-    const stream1 = await TileDocument.create<any>(ceramic, INITIAL_CONTENT, { deterministic: true }, { syncTimeoutSeconds: 0 });
-    await anchorUpdate(ceramic, stream1);
+    const metadata = { deterministic: true, family: "family", tags: ["x", "y"], controllers: controllers}
+    const stream1 = await TileDocument.create<any>(ceramic, null, metadata, { syncTimeoutSeconds: 0 });
```

Comment:
maybe add `anchor:false` to the opts here if the test isn't going wait for the anchor to succeed. Otherwise the test can fail nondeterministically due to a race between the anchor and the update

Reply:
What's the right principle to follow when calling `anchorUpdate()`? Does your comment apply just to this one test case or in more places where `anchorUpdate()` is being called?

Reply:
`anchorUpdate` works bests for tests that have their ceramic node set to `anchorOnRequest: false`, since `anchorUpdate` [instructs the anchor service](https://github.com/ceramicnetwork/js-ceramic/blob/62e87b19d36c9ce8dce76323f61004980c030b6e/packages/core/src/state-management/__tests__/anchor-update.ts#L14) to perform an on-demand anchor.  So the `anchor` opt to `TileDocument{create,update}` controls whether that specific operation sends an anchor request to the anchor service.  If `anchorOnRequest` is set to true (the default behavior), then the anchor service will automatically perform an anchor at some point after receiving the request. If `anchorOnRequest` is false, then the anchor service will only perform the anchor if the `anchor()` method is explicitly called against the anchor service.

So the combination of `anchorOnRequest: false` + using `anchorUpdate` (which manually triggers the anchor), is the current best practice for writing reliably deterministic tests that perform anchors. There are still, however, lots of tests out there that are not written so deterministically, so you need to be careful when copy+pasting from or updating existing tests.

Reply:
Thanks for explaining!!